packages/todo-bot/src/scanner.ts

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GCFLogger, logger as defaultLogger} from 'gcf-utils';
+import * as child_process from 'child_process';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+export interface Annotation {
+  name: string;
+  file: string;
+  line: number;
+  preview: string;
+  context?: string;
+}
+
+interface ScannerOptions {
+  logger?: GCFLogger;
+  workdir?: string;
+}
+
+export class AnnotationScanner {
+  readonly owner: string;
+  readonly repo: string;
+  readonly branch: string;
+  private logger: GCFLogger;
+  private workdir?: string;
+
+  constructor(
+    owner: string,
+    repo: string,
+    branch: string,
+    options?: ScannerOptions
+  ) {
+    this.owner = owner;
+    this.repo = repo;
+    this.branch = branch;
+    this.logger = options?.logger ?? defaultLogger;
+    this.workdir = options?.workdir;
+  }
+
+  private async getWorkdir(): Promise<string> {
+    if (!this.workdir) {
+      this.workdir = fs.mkdtempSync(
+        path.join(os.tmpdir(), `${this.owner}-${this.repo}`)
+      );
+      this.logger.info(
+        `Cloning ${this.owner}/${this.repo} (${this.branch}) into ${this.workdir}`
+      );
+      await exec(
+        `git clone --depth=1 --branch=${this.branch} https://github.com/${this.owner}/${this.repo}.git ${this.workdir}`,
+        {logger: this.logger}
+      );
+    }
+    return this.workdir;
+  }
+
+  // TODO: example todo
+  // TODO(context): another example annotation
+  async findAnnotations(annotations: string[]): Promise<Annotation[]> {
```

Comment:
How about to make the interface receiving a regex and returning the match object?

That way it's more generic